src/code/audio/audio_synthesis.c

```diff
@@ -1,137 +1,1705 @@
+    curVolRight = synthState->curVolRight;
+
+    targetVolLeft = sampleState->targetVolLeft;
+    targetVolLeft <<= 4;
+    targetVolRight = sampleState->targetVolRight;
+    targetVolRight <<= 4;
+
+    if ((gAudioContext.soundMode == SOUNDMODE_SURROUND_EXTERNAL) && (sampleState->surroundEffectIndex != 0xFF)) {
+        defaultPanVolume = gDefaultPanVolume[sampleState->surroundEffectIndex];
+        targetVolLeft *= defaultPanVolume;
+        targetVolRight *= defaultPanVolume;
+    }
+
+    if (targetVolLeft != curVolLeft) {
+        rampLeft = (targetVolLeft - curVolLeft) / (numSamplesPerUpdate >> 3);
+    } else {
+        rampLeft = 0;
+    }
+
+    if (targetVolRight != curVolRight) {
+        rampRight = (targetVolRight - curVolRight) / (numSamplesPerUpdate >> 3);
+    } else {
+        rampRight = 0;
+    }
+
+    curReverbVolAndFlags = synthState->curReverbVol;
+    curReverbVol = curReverbVolAndFlags & 0x7F;
+
+    if (curReverbVolAndFlags != targetReverbVol) {
+        rampReverb = (((targetReverbVol & 0x7F) - curReverbVol) << 9) / (numSamplesPerUpdate >> 3);
+        synthState->curReverbVol = targetReverbVol;
+    } else {
+        rampReverb = 0;
+    }
+
+    synthState->curVolLeft = curVolLeft + (rampLeft * (numSamplesPerUpdate >> 3));
+    synthState->curVolRight = curVolRight + (rampRight * (numSamplesPerUpdate >> 3));
+
+    if (sampleState->bitField1.useHaasEffect) {
+        AudioSynth_ClearBuffer(cmd++, DMEM_HAAS_TEMP, DMEM_1CH_SIZE);
+        AudioSynth_EnvSetup1(cmd++, curReverbVol * 2, rampReverb, rampLeft, rampRight);
+        AudioSynth_EnvSetup2(cmd++, curVolLeft, curVolRight);
+
+        switch (haasEffectDelaySide) {
+            case HAAS_EFFECT_DELAY_LEFT:
+                // Store the left dry channel in a temp space to be delayed to produce the haas effect
+                dmemDests = sEnvMixerLeftHaasDmemDests;
+                break;
+
+            case HAAS_EFFECT_DELAY_RIGHT:
+                // Store the right dry channel in a temp space to be delayed to produce the haas effect
+                dmemDests = sEnvMixerRightHaasDmemDests;
+                break;
+
+            default: // HAAS_EFFECT_DELAY_NONE
+                dmemDests = sEnvMixerDefaultDmemDests;
+                break;
+        }
+    } else {
+        aEnvSetup1(cmd++, curReverbVol * 2, rampReverb, rampLeft, rampRight);
+        aEnvSetup2(cmd++, curVolLeft, curVolRight);
+        dmemDests = sEnvMixerDefaultDmemDests;
+    }
+
+    aEnvMixer(cmd++, dmemSrc, numSamplesPerUpdate, (curReverbVolAndFlags & 0x80) >> 7,
+              sampleState->bitField0.strongReverbRight, sampleState->bitField0.strongReverbLeft,
+              sampleState->bitField0.strongRight, sampleState->bitField0.strongLeft, dmemDests, sEnvMixerOp);
+
+    return cmd;
+}
+
+Acmd* AudioSynth_LoadWaveSamples(Acmd* cmd, NoteSampleState* sampleState, NoteSynthesisState* synthState,
+                                 s32 numSamplesToLoad) {
+    s32 numSamplesAvail;
```

Comment:
I think at least leave a comment that this is Available since Avail is both a word in its own right and quite close to the character's name (yes I know it should make sense in context but I'm thinking about people searching the repo).